### PR TITLE
Editor: Unify the list view shortcut registration and definition

### PR DESCRIPTION
**Pull request by youknowriad**
Related #52632 

## What?

This is a small refactor to unify the list view keyboard shortcut definition between the site and post editors, since now the state has moved to the editor package.

## Testing Instructions

1- You can open and close the list view using the shortcut

### CODE REVIEW

File: packages/edit-post/src/components/keyboard-shortcut-help-modal/test/index.js [hunk 1]
```diff
@@ -16,21 +16,6 @@ import { KeyboardShortcutHelpModal } from '../index';
...
-				<EditorKeyboardShortcutsRegister />
-				<KeyboardShortcutHelpModal isModalActive toggleModal={ noop } />
-			</>
-		);
-
-		expect(
-			screen.getByRole( 'dialog', {
-				name: 'Keyboard shortcuts',
-			} )
-		).toMatchSnapshot();
```
Comment (youknowriad) [inline packages/edit-post/src/components/keyboard-shortcut-help-modal/test/index.js:31]: This is a useless test (we just update the snapshot when it changes)

**Comment by Mamaduka** [comment]
I think the `edit-site` package doesn't register global shortcuts from the editor package (`EditorKeyboardShortcuts`); this is the reason e2e tests are failing.

Opportunity for more unification 😄

**Review by andrewserong — COMMENTED** [review]
> I think the edit-site package doesn't register global shortcuts from the editor package (EditorKeyboardShortcuts); this is the reason e2e tests are failing.

Thanks @Mamaduka, that makes sense! And it's consistent with manual testing (CTRL+Option+o opens the list view in the post editor but not the site editor for me with this PR applied).

File: packages/edit-post/src/components/keyboard-shortcuts/index.js [hunk 2]
```diff
@@ -101,16 +99,6 @@ function KeyboardShortcuts() {
 			},
 		} );
 
-		registerShortcut( {
-			name: 'core/edit-post/toggle-list-view',
```
Comment (ellatrix) [inline packages/edit-post/src/components/keyboard-shortcuts/index.js:105]: I guess unregistering this won't work anymore? Anyway, I think that's fine :)

**Review by ellatrix — APPROVED** [review]
Works :)

**Pull request merged**